docs/only-trusted.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Signing Test Firmware Payloads
+---
+
+## Introduction
+
+In the normal vendor update flow, firmware is optionally signed and encrypted by the vendor, and
+then uploaded to the LVFS wrapped in a cabinet archive with a small XML metadata file to describe
+how the firmware should be matched to hardware.
+
+Upon uploading, the LVFS signs the firmware and metadata XML contained in the archive and adds it to
+a `.jcat` file also included in the image.
+The original firmware is never modified, which is why the [Jcat](https://github.com/hughsie/libjcat)
+file exists as both a detached checksum (SHA-1, SHA-256 and SHA-512) and a detached signature.
+The LVFS can add either GPG or PKCS#7 signatures in the Jcat file, and currently does *both* for
+maxmum compatibility with how client systems have been configured.
+
+The keys in `/etc/pki/fwupd` and `/etc/pki/fwupd-metadata` are used for per-system trust and are
+currently used for "did the firmware update *come from* somewhere I trust" rather than "verify the
+vendor signed the update" -- on the logic the "signed the update" is already covered by the signature
+on the EFI capsule payload.
+Notably, The LVFS both verifies the vendor OEM -> ODM -> IHV relationships and assigns restrictions
+on what devices each legal entity can upload for.
+
+There's no way to separate the keys so that you could say "only use this certificate for
+per-system-trust when the DMI vendor of the device is Dell" and there's no way to do key rotation
+or revocation.
+The trusted certificate mechanism was not really designed for any keys except the static LVFS.
+
+If the intent is to use a test key to sign the firmware files and get installed purely offline with
+an unmodified fwupd package (without uploading to the LVFS) then the following instructions can be
+modified to suit.
+
+First, lets verify that an existing firmware binary and metainfo file without a Jcat signature
+refuses to install when packaged into a cabinet archive:
+
+    $ gcab -c firmware.cab firmware.bin firmware.metainfo.xml 
+    $ fwupdmgr install firmware.cab --allow-reinstall
+    Decompressing…           [ -                                     ]
+    firmware signature missing or not trusted; set OnlyTrusted=false in /etc/fwupd/daemon.conf ONLY if you are a firmware developer
```

Comment:
another way to do is is just `fwupdtool install-blob firmware.bin`.  Should that be a way to suggest too?

Reply:
Yeah I'm mostly thinking to mention it because sometimes you don't need the signing infrastructure but rather just the ability to install payloads.